feat: Adds to charms best practices concerning respecting model-config  settings (#20268)
juju/juju: docs/reference/charm/charm-maturity.md

| Newly proposed relations have been reviewed and approved by experts to ensure:<p>&#8226; The relation is ready for adoption by other charmers from a development best practice point of view.<p>&#8226;  No conflicts with existing relations of published charms.<p>&#8226;  Relation naming and structuring are consistent with existing relations.<p>&#8226; Tests cover integration with the applications consuming the relations. | A [Github project](https://github.com/canonical/charm-relation-interfaces) structures and defines the implementation of relations.<p>No new relation should conflict with the ones covered by the relation integration set [published on Github](https://github.com/canonical/charm-relation-interfaces).<p>&#8226; See more: [Charmcraft | Manage relations](https://canonical-charmcraft.readthedocs-hosted.com/stable/howto/manage-charms/#manage-relations), [Ops | Manage relations](https://ops.readthedocs.io/en/latest/howto/manage-relations.html)|

#### The charm respects juju model-config

> Most developers are keenly aware of their own charm's configs, without being aware that juju model-config is another point of administrative control.

Reviewer: ```suggestion
Most developers are keenly aware of their own charm's configs, without being aware that `juju model-config` is another point of administrative control.
```

| Objectives  | Tips, examples, further reading |
| ----------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------- |

> | Avoid duplicating configuration options that is best controlled at a model perspective: <p>&#8226; juju-http-proxy, juju-https-proxy, juju-no-proxy should influence the charms behavior when the charm or charmed application makes any http request. | A [Github project](https://github.com/canonical/charms.proxylib) provides a library to help charms direct url requests and subprocess calls through the model configured proxy environment. |

Reviewer: ```suggestion
| Avoid duplicating configuration options that are best controlled at a model level: <p>&#8226; `juju-http-proxy`, `juju-https-proxy`, `juju-no-proxy` should influence the charm's behavior when the charm or charmed application makes any HTTP request. | A [Github project](https://github.com/canonical/charms.proxylib) provides a library to help charms direct URL requests and subprocess calls through the model-configured proxy environment. |
```
